frontend/src/pages/articles/components/article-card/article-card.tsx

```diff
@@ -0,0 +1,122 @@
+import ArticlePreview from '~/assets/img/article-preview.png';
+import { Avatar, Icon, Link, Tag } from '~/libs/components/components.js';
+import { AppRoute } from '~/libs/enums/enums.js';
+
+import {
+  type ArticleType,
+  type TagType,
+  type UserAuthResponseDto,
+} from './libs/types/types.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  article: ArticleType;
+  user: UserAuthResponseDto;
+};
+
+const MOCKED_TEXT: string = `The developer technology landscape changes all the time as new tools
+  and technologies are introduced. After having lots of interviews and
+  reading through countless job descriptions on job boards I think this
+  is a great modern tech stack for JavaScript developers in 2021.`;
+
+const MOCKED_TAGS: TagType[] = [
+  { id: 1, name: 'IT' },
+  { id: 2, name: 'CODE' },
+  { id: 3, name: 'Humor' },
+  { id: 4, name: 'Work' },
+  { id: 5, name: 'Tech' },
+];
+
+const MOCKED_ARTICLE: ArticleType = {
+  publishedAt: 'May 28',
+  timeSincePublication: '7 min read',
+  title: 'Modern Full-Stack Developer Tech Stack 2021',
+  text: MOCKED_TEXT,
+  comments: '540',
+  views: '367K',
+  likes: '36K',
+  dislikes: '18K',
+  tags: MOCKED_TAGS,
+};
+
+const MOCKED_USER: UserAuthResponseDto = {
+  id: 7,
+  email: 'nolanaris@gmail.com',
+  firstName: 'Nolan',
+  lastName: 'Saris',
+};
+
+const ArticleCard: React.FC<Properties> = ({
+  article = MOCKED_ARTICLE,
+  user = MOCKED_USER,
+}) => {
+  const {
+    publishedAt = 'May 28',
+    timeSincePublication = '7 min read',
+    title = 'Modern Full-Stack Developer Tech Stack 2021',
+    text = MOCKED_TEXT,
+    comments = '540',
+    views = '367K',
+    likes = '36K',
+    dislikes = '18K',
+    tags: propertyTags,
+  } = article;
+
+  const { firstName, lastName } = user;
+  const fullName = `${firstName} ${lastName}`;
+  const tags = propertyTags.map((tag) => <Tag key={tag.id} name={tag.name} />);
+
+  return (
+    <article className={styles.article}>
+      <div className={styles.header}>
+        <div className={styles.info}>
+          <Avatar username={fullName} avatarUrl={null} />
+          <span className={styles.publisherName}>{fullName}</span>
+          <span className={styles.publicationTime}>{publishedAt}</span>
+          <span className={styles.publicationTime}>{timeSincePublication}</span>
+        </div>
+        <Icon iconName="favorite" className={styles.favoriteIcon} />
+      </div>
+      <div className={styles.body}>
+        <div>
+          <h4 className={styles.title}>{title}</h4>
+          <p className={styles.text}>{text}</p>
+          <div className={styles.tags}>{tags}</div>
+        </div>
+        <img
+          src={ArticlePreview}
+          alt="article preview"
+          width="134"
+          height="134"
```

Comment:
Apply width and height restrictions from class styling, instead of directly

Reply:
Isn't it good practice to define width and height for images to reserve space on page load?

Reply:
Yep, but as of now we'd better care about responsiveness